src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -375,19 +380,30 @@ int xma_xclbin_info_get(const char *buffer, XmaXclbinInfo *info)
     return XMA_SUCCESS;
 }
 
-int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank)
+int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank, bool has_mem_grps)
 {
     //64 bits based on MAX_DDR_MAP = 64
     int ddr_bank_idx = 0;
-    while (bit_map != 0)
-    {
-        if (bit_map & 1)
-        {
-            *ddr_bank = ddr_bank_idx;
-            return XMA_SUCCESS;
+    if (!has_mem_grps) {
+        while (bit_map != 0) {
+            if (bit_map & 1) {
+                *ddr_bank = ddr_bank_idx;
+                return XMA_SUCCESS;
+            }
+            ddr_bank_idx++;
+            bit_map = bit_map >> 1;
+        }
+    } else {//For Memory Groups use last group as default memory group; For HBM groups
+        ddr_bank_idx = 63;
+        uint64_t tmp_int = 1ULL << 63;
+        while (bit_map != 0) {
+            if (bit_map & tmp_int) {
+                *ddr_bank = ddr_bank_idx;
+                return XMA_SUCCESS;
+            }
+            ddr_bank_idx--;
+            bit_map = bit_map << 1;
```

Comment:
Ideally you should always search from the back, but given that I have seen a decoder example where the xclbin was modified to manually expand the *first* bank entry, searching from back would break this use case whether we deem it valid or not.

Reply:
But why your wisdom is not shown in some comments???